test/common.bash

```diff
@@ -81,11 +81,33 @@ image_ok () {
 
 multiprocess_ok () {
     [[ $ch_multiprocess ]] || skip 'no multiprocess launch tool found'
-    # If the MPI in the container is MPICH, we only try host launch on Crays.
-    # For the other settings (workstation, other Linux clusters), it may or
-    # may not work; we simply haven't tried.
-    [[ $ch_mpi = mpich && -z $ch_cray ]] \
-        && skip 'MPICH untested'
+
+    # MPICH requires different handling from OpenMPI. Set a variable to enable
+    # some kludges.
+    if [[ $1 = *'-mpich' ]]; then
+        # If the MPI in the container is MPICH, we only try host launch on Crays.
+        # For the other settings (workstation, other Linux clusters), it may or
+        # may not work; we simply haven't tried.
+        [[ -z $ch_cray ]] && skip 'MPICH untested'
+        ch_mpi=mpich
+        # First kludge. MPICH's internal launcher is called "Hydra". If Hydra sees
+        # Slurm environment variables, it tries to launch even local ranks with
+        # "srun". This of course fails within the container. You can't turn it off
+        # by building with --without-slurm like OpenMPI, so we fall back to this
+        # environment variable at run time.
+        export HYDRA_LAUNCHER=fork
+    else
+        ch_mpi=openmpi
+    fi
+
+    # OpenMPI 3.1 pukes when guest-launched and Slurm environment variables
+    # are present. Work around this by fooling OpenMPI into believing it's not
+    # in a Slurm allocation.
+    if [[ $ch_mpi = openmpi ]]; then
+        # shellcheck disable=SC2034
+        ch_unslurm='--unset-env=SLURM*'
+    fi
+
```

Comment:
What's the reasoning to move this stuff? It significantly expands what the function does, and the other `_ok` functions don't have side effects beyond skipping, do they?